cmd/shellNode.go

```diff
@@ -0,0 +1,225 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2instanceconnect"
+
+	"github.com/instructure-bridge/truss-cli/truss"
+)
+
+var shellNodeCmd = &cobra.Command{
+	Use:   "node [node-name]",
+	Short: "Launch a shell on a Truss node.",
+	Long: `Launch a shell on a Truss node via SSH.
+
+	Usage:
+
+	truss -e [env] shell node [instance-id]
+	truss -e [env] shell node [node-name]
+
+	Examples:
+
+  truss -e nonprod-cmh shell node i-0b35c43cd48ab85ee
+  truss -e nonprod-cmh shell node ip-10-12-11-61.us-east-2.compute.internal`,
+	Args: cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		region, err := getAWSRegionFromKubeconfig()
+		if err != nil {
+			return errors.Wrap(err, "Failed to retrieve AWS region from Kubeconfig")
+		}
+
+		awsRole := viper.GetString("aws.assumeRole")
+		if awsRole == "" {
+			return errors.New("Unable to find aws.assumeRole in config file")
+		}
+
+		sess := truss.NewAWSSession(region, awsRole)
+
+		var instanceID string
+
+		if strings.HasPrefix(args[0], "ip-") {
+			// Lookup instance-id from Kubernetes if given a node name
+			kubeNode, err := describeKubernetesNode(args[0])
+			if err != nil {
+				return errors.Wrap(err, "Unable to describe Kubernetes node")
+			}
+
+			providerIDParts := strings.Split(kubeNode.Spec.ProviderID, "/")
+			instanceID = providerIDParts[len(providerIDParts)-1]
+		} else if strings.HasPrefix(args[0], "i-") {
+			instanceID = args[0]
+		}
+
+		instance, err := describeInstance(instanceID, sess)
+		if err != nil {
+			return errors.Wrap(err, "Unable to describe EC2 instance")
+		}
+
+		availabilityZone := *instance.Placement.AvailabilityZone
+		hostname := *instance.PrivateDnsName
+
+		publicKey, err := getSSHPublicKey()
+		if err != nil {
+			return err
+		}
+
+		username := "ec2-user"
```

Comment:
Should we add a `-u` flag for this that defaults to `ec2-user`?

Reply:
Done and tested. `ec2-user` has full sudo privileges, so I don't think this is something that will actually be used often, but an excuse to play with a flag. :)